server/resp/src/main/java/org/infinispan/server/resp/CacheCommandRequest.java

```diff
@@ -0,0 +1,17 @@
+package org.infinispan.server.resp;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.AdvancedCache;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+public record CacheCommandRequest (CompletionStage<RespRequestHandler> myStage,
```

Comment:
We should remove this and add an interface that the BaseRespDecoder implements so you can just pass the decoder instance as the argument instead. It avoids this allocation.